### ISSUE TITLE
can efficientNMS plugin get dynamic outputs?

### ISSUE THREAD

**Issue by jike-algorithm-zhangxiao** [NVIDIA/TensorRT#2074]
## Description
I convert onnx model which has efficientNMS plugin to tensorrt, and i deploy it use triton.  
But when starting inference, the scores shape is always [max_output_boxes, 1, 1], the boxes shape is always [max_output_boxes, 1, 4]. 
Even not have targert objects, the output will padded the box which has max scores to [max_output_boxes, ..]. Why like this? shall I have some insights to do for this?

## Environment

**TensorRT Version**: 8.2.1.8
**NVIDIA GPU**: T4
**NVIDIA Driver Version**: 460.73.01 
**CUDA Version**: 11.2
**CUDNN Version**: 8.2
**Operating System**: Ubuntu
**Python Version (if applicable)**: 3.8
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 1.10
**Baremetal or Container (if so, version)**: 22.01

**Comment by wraveane** [comment]
Hi @jike-algorithm-zhangxiao , yes, you are correct. The current implementation of the EfficientNMS plugin is limited in that way, as TensorRT currently only supports fixed tensor shapes. So for now, you will unfortunately need to remove the "padding" detections in a post-processing step in host code (either python or c++) after Triton performs inference. One of the output tensors of the plugin is "num_detections", which holds the number of valid detections found for each image. So using these values, you would only process the first N scores, the first N classes, and the first N boxes for each image.

This is something that may be improved in the future, but we don't have a specific timeline for that, so for now, it's better to do this post-processing step to ignore the padding detections.

**Comment by jike-algorithm-zhangxiao** [comment]
ok, Thanks!
